src/wasm/wasm-type.cpp

```diff
@@ -680,6 +684,14 @@ std::optional<Type> TypeInfo::getCanonical() const {
           case HeapType::i31:
           case HeapType::data:
             break;
+          case HeapType::string:
+            return Type(HeapType::string, Nullable);
+          case HeapType::stringview_wtf8:
+            return Type(HeapType::stringview_wtf8, Nullable);
+          case HeapType::stringview_wtf16:
+            return Type(HeapType::stringview_wtf16, Nullable);
+          case HeapType::stringview_iter:
+            return Type(HeapType::stringview_iter, Nullable);
```

Comment:
Just `break` here so that `getCanonical` ends up returning `{}` in these cases and you should be good to go.

Reply:
Thanks! Right, so `{}` means it doesn't emit a canonical thing here, I guess, which is what we want... works for me locally, updated.